### PR TITLE
Adding SoC tests

### PR DESCRIPTION
**Pull request by mithro**
This pull request adds two tests to the iCE40 architectures. They are;
 * PicoSoC - A small SoC based on the PicoRV32.
 * Murax - A small SoC based on the VexRISC core.

### CODE REVIEW

File: make/tests.mk [hunk 1]
```diff
@@ -113,56 +125,56 @@ rr_graph.xml: rr_graph.real.xml
...
+else
+SOURCE_E = $(filter %.eblif,$(SOURCES))
+SOURCE_V = $(filter %.v,$(SOURCES))
+endif
 
-ifneq ($(words $(SOURCES)),1)
 ifeq ($(words $(SOURCES)),0)
 $(error "No sources found!")
 endif
-$(error "Multiple sources found! $(SOURCES)")
```
Comment (elmsfu) [inline make/tests.mk]: Should we add a check for the case of multiple eblif files?

**Comment by mithro** [comment]
I added a `README.md` for the Murax case. Do you have the same details for PicoSoC?

File: make/tests.mk (hunk 1 above)
Comment (mithro) [reply make/tests.mk]: Done.

**Comment by mithro** [comment]
So this is currently producing;
```
symbiflow-arch-defs/ice40/make/boards.mk:5: warning: overriding recipe for target 'symbiflow-arch-defs/third_party/icestorm//iceprog/iceprog'
symbiflow-arch-defs/ice40/make/tools.mk:6: warning: ignoring old recipe for target 'symbiflow-arch-defs/third_party/icestorm//iceprog/iceprog'
OUT_LOCAL = symbiflow-arch-defs/ice40/tests/picosoc//build-ice40--hx8k
OUT_LOCAL = symbiflow-arch-defs/ice40/tests/picosoc//build-ice40-top-routing-virt-hx8k
```

I assume because I pulled in your splitting incorrectly?

**Pull request merged**